modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -327,39 +384,241 @@ public function insertOrderToDb($id_order, $address, $value, $bits) {
-	public function getOrderByAddress($bitcoinAddress) {
+	public function newOrder($amount, $id_order) {
+
 		try {
 			$existing_order = Capsule::table('blockonomics_bitcoin_orders')
-				->where('addr', $bitcoinAddress)
+				->where('id_order', $id_order)
+				->where('addr', '')
+				->where('value', $amount)
+				->value('id_order');
+		} catch (\Exception $e) {
+				echo "Unable to select order from blockonomics_bitcoin_orders: {$e->getMessage()}";
+		}
+
+		if(!$existing_order) {
+			try {
+				Capsule::table('blockonomics_bitcoin_orders')->insert(
+					[
+						'id_order' => $id_order,
+						'timestamp' => time(),
+						'status' => -1,
+						'value' => $amount
+					]
+				);
+			} catch (\Exception $e) {
+					echo "Unable to insert new order into blockonomics_bitcoin_orders: {$e->getMessage()}";
+			}
+		}		
+
+		$crypted_id = $this->encrypt($id_order);
+
+		return $crypted_id;
+	}
+
+	/*
+	 * Get the order id using the order uuid
+	 */	
+	public function getOrderIdByHash($order_hash) {
+		$order_id = $this->decrypt($order_hash);
+		try {
+			$first_order = Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
 				->orderBy('timestamp', 'desc')
 				->first();
 		} catch (\Exception $e) {
-				echo "Unable to select order from blockonomics_bitcoin_orders: {$e->getMessage()}";
+				echo "Unable to get order id from blockonomics_bitcoin_orders: {$e->getMessage()}";
 		}
 
-		$row_in_array = array(
-			"id" => $existing_order->id,
-			"order_id" => $existing_order->id_order,
-			"timestamp"=> $existing_order->timestamp,
-			"status" => $existing_order->status,
-			"value" => $existing_order->value,
-			"bits" => $existing_order->bits,
-			"bits_payed" => $existing_order->bits_payed
-		);
+		return $first_order->id_order;
+	}
 
-		return $row_in_array;
+	/*
+	 * Get all orders linked to uuid
+	 */	
+	public function getAllOrdersByHash($order_hash) {
+		$order_id = $this->decrypt($order_hash);
+		try {
+			return Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
+				->orderBy('timestamp', 'desc')->get();
+		} catch (\Exception $e) {
+				echo "Unable to get order id from blockonomics_bitcoin_orders: {$e->getMessage()}";
+		}
+		return;
+	}
+
+	/*
+	 * Get client currency by invoice id
+	 */	
+	public function getClientCurrency($order_hash) {
+		$order_id = $this->decrypt($order_hash);
+		$all_orders_by_id = Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
+				->orderBy('timestamp', 'desc')->get();
+		return $all_orders_by_id;
+	}
+
+	/*
+	 * Check for pending orders and return if exists
+	 */	
+	public function isOrderPending($orders) {
+		foreach ($orders as $order) {
+			//check if status 0 or 1
+			if($order->status == 0 || $order->status == 1){
+				return $order;
+			}
+		}
+		return;
+	}
+
+	/*
+	 * Check for unused address for the currency linked to the order
+	 */		
+	public function isOrderWaiting($orders, $blockonomics_currency) {
```

Comment:
Again.. rename function since does also does updates and not only doing a get. Something like *getAndUpdateWaitingOrder*

Reply:
Updated in 18ea106